gm4_blast_furnaces/data/blast_furnaces/functions/check_ore.mcfunction

```diff
@@ -0,0 +1,10 @@
+# @s = blast furnace that is looking at a lit furnace
+# run from processes
+
+data modify entity @s HandItems[0] set from block ^ ^ ^1 Items[{Slot:0b}]
+replaceitem entity @s armor.head air
+execute store result score can_smelt gm4_bf_data run loot replace entity @s armor.head fish blast_furnaces:smeltable_display ~ ~ ~ mainhand
+replaceitem entity @s weapon air
+
+#check if structure is valid
+execute if score can_smelt gm4_bf_data matches 1.. if block ^ ^1 ^1 glass if block ~1 ~ ~1 iron_block if block ~1 ~ ~-1 iron_block if block ~-1 ~ ~1 iron_block if block ~-1 ~ ~-1 iron_block if block ~1 ~1 ~1 iron_block if block ~1 ~1 ~-1 iron_block if block ~-1 ~1 ~1 iron_block if block ~-1 ~1 ~-1 iron_block if block ^ ^ ^-1 iron_block if block ^1 ^ ^ iron_block if block ^-1 ^ ^ iron_block if block ^ ^1 ^-1 iron_block if block ^1 ^1 ^ iron_block if block ^-1 ^1 ^ iron_block run function blast_furnaces:check_cook_time
```

Comment:
Glass is not a required part of the blast furnace structure (to allow a hopper to feed in ores)

Reply:
It should probably also check that the space above the hopper is air as some blocks above the hopper would stop the hopper's ability to suck up items.

Reply:
oof, I haven't added that